Scripts/End/EndPlatformController.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EndPlatformController : MonoBehaviour
+{
+
+    [Space, SerializeField]
+    GameObject endPlatform;
+
+    [Space, SerializeField]
+    ParticleSystem smoke;
+
+    [Space, SerializeField]
+    ParticleSystem fire;
+    Collider player;
+
+
+    private void OnTriggerEnter(Collider other)
+    {
+        player = other;
+        PlayerMovement playerMovement = other.GetComponent<PlayerMovement>();
+        playerMovement.setIsGameEnd(true);
+
+        GameObject environment = endPlatform.transform.parent.gameObject;
+        Rigidbody rigidbody = environment.GetComponent<Rigidbody>();
+        rigidbody.constraints = RigidbodyConstraints.FreezePositionZ;
+
+        CheckWater();
+    }
+
+    private void CheckWater()
+    {
+        PlayerStats playerStats = player.GetComponent<PlayerStats>();
+        double water = playerStats.getWater();
```

Comment:
Yukarıda player collider ile trigger'lanmazsa, PlayerStats değeri null olur ve oyun crash olur. O yüzden null check eklenmesi gerekir.

```suggestion
        PlayerStats playerStats = player.GetComponent<PlayerStats>();
        double water;
        
        if (playerStats)
        {
            water = playerStats.getWater();
        }
```